llarp/service/endpoint.cpp

```diff
@@ -1252,23 +1252,25 @@ namespace llarp
     bool
     Endpoint::ShouldBuildMore(llarp_time_t now) const
     {
+      static constexpr auto buildSpread = path::default_lifetime / 4;
       if(path::Builder::BuildCooldownHit(now))
         return false;
-      const bool should = path::Builder::ShouldBuildMore(now);
+      const bool shouldBuild = NumPathsExistingAt(now + buildSpread) < numPaths
+          and NumInStatus(path::ePathBuilding) < numPaths;
```

Comment:
This can just `return true` early if the condition is satisfied; nothing done below can ever result in anything other than a true return value.

Reply:
you're right.